### ISSUE TITLE
`InetAddress#getCanonicalHostName` for unknown IP address throws `UndefinedBehaviorError`

### ISSUE THREAD

**Issue by armanbilge** [scala-native/scala-native#2804]
```scala
@main def main =
  println(java.net.InetAddress.getByName("240.0.0.0").getCanonicalHostName)
```

```
$ scala-cli bug.scala 
240.0.0.0
```

```
$ scala-cli --native bug.scala 
scala.scalanative.runtime.UndefinedBehaviorError
        at java.lang.StackTrace$.currentStackTrace(Unknown Source)
        at java.lang.Throwable.fillInStackTrace(Unknown Source)
        at scala.scalanative.runtime.package$.throwUndefined(Unknown Source)
        at <none>.(Unknown Source)
        at main.main(Unknown Source)
        at <none>.main(Unknown Source)
        at <none>.__libc_start_main(Unknown Source)
        at <none>._start(Unknown Source)
```

**Comment by LeeTibbert** [comment]
@arman   

Sorry, but I can not replicate, scala 3.13, SN  0.5.0-SNAPSHOT.  I get link time missing symbols.
When I grep all `*.scala` for `getCanonicalHostName`  I get nothing found.

Are you running code from one of your recent PRs?

**Comment by armanbilge** [comment]
@LeeTibbert thanks for giving it a try. Huh, that's strange: I'm literally running this example in a clean sandbox with scala-cli against Native 0.4.5. Those are my exact code and exact commands.

It's a minimization of an error I was getting in this branch:
https://github.com/armanbilge/ip4s/tree/feature/native

**Comment by armanbilge** [comment]
But you are right, I completely forgot that I added the `getCanonicalHostName` method in https://github.com/scala-native/scala-native/pull/2752.

What I don't understand is why the code is linking here and then throwing `UndefinedBehaviorError`. But maybe this is a moot issue.

**Comment by LeeTibbert** [comment]
@arman   

re: Are you running code from one of your recent PRs?

Must be.  If yes, then sorry but you are hosed.  Javalib IPv6 flat out
does not work outside of PR #2803, which I created today.   I can say
that with some confidence.

I took a quick look at your `InetAddress` changes PR. Looks like
I will have to pickup those changes when time comes to submit
@2803 as a full fledged PR.

**Comment by armanbilge** [comment]
@LeeTibbert forgive an ignorant question, how does this relate to ipv6?

**Comment by LeeTibbert** [comment]
Given your model, someday I am going to have to learn/try `scala-cli`

**Comment by LeeTibbert** [comment]
I can take a deeper look, but the GitHub diff in your PR mentioned `ipv6` several places.  In theory,
that code should never have executed. I was about to go trackdown how you determined `ipv6`.
Hard to follow clever people with on-sight reading....

I can trace our PR deeper and/or I can try it in an environment where SN IPv6 support works.

Seems like `getCannonicalHostName()` would be useful, especially if it is specified by Java 8.

**Comment by LeeTibbert** [comment]
In your `getCannonicalHostName` PR, the change in `SocketHelpers.scala` from `sizeof[sockaddr].toUInt,` to
```
if (isV6) sizeof[sockaddr_in6].toUInt
            else sizeof[sockaddr_in].toUInt,
```
because a `sockaddr` is defined/allocated to be large enough to be either (or one of a set of a bunch of others).


That should not be causing this problem. 

Since I will eventually have to be merging that PR, I hope, want me to set up a `0.4.x` environment and
figure it out?  Easier to do it using only IPv4.

**Comment by LeeTibbert** [comment]
Running reproducer in SN build with fresh 0.4.x clone & a fresh ~/.ivy2/local, 
sandbox3/run gives:

```
[info] Total (17780 ms)
240.0.0.0
[success] Total time: 248 s (04:08), completed Aug 28, 2022, 3:35:21 PM
```

I can try to figure out how to download scala-cli and try that.

If asked, I would say that the SN Inet*Test files exercised at least
one bogus IPv4 address.  I would have to check again before asserting
that.

**Comment by armanbilge** [comment]
@LeeTibbert thanks! I think that confirms that https://github.com/scala-native/scala-native/pull/2752 fixes this issue so I will close.

**Comment by LeeTibbert** [comment]
Good thing you did #2752 then...
